src/browser/search.py

```diff
@@ -23,3 +24,13 @@ def search(self, query):
     def get_first_link(self):
         return self.query_result["webPages"]["value"][0]["url"]
 
+class DuckDuckGoSearch:
+    def search(self, query):
+        try:
+            self.query_result = DDGS().text(query, max_results=5)
+            return self.query_result
+        except Exception as err:
+            return err
+
+    def get_first_link(self):
+        return self.query_result[0]["href"]
```

Comment:
**Suggestion**: Implement error handling for the DuckDuckGoSearch class to manage cases where the DDGS API might fail or return unexpected responses. This will enhance the robustness of the search functionality. <br> **Code Suggestion**: 
 ```
 +        try:
 +            # Existing code for DDGS().text(query, max_results=5)
 +        except Exception as e:
 +            # Log error and potentially fallback to another search engine or return a meaningful error message
 ```